oct/cli/image_ready.py

```diff
@@ -0,0 +1,103 @@
+# coding=utf-8
+from __future__ import absolute_import, division, print_function
+
+import sys
+
+from click import command, pass_context, ClickException, echo, option, UsageError
+
+from .util.cloud_provider.image_options import operating_system_option, stage_option, ami_id_option
+from .util.cloud_provider.common_options import Provider, provider_option
+
+import boto3
+from datetime import datetime
+
+_SHORT_HELP = 'Check if the specified or most recent matching image is tagged ready.'
+
+
+@command(
+    short_help=_SHORT_HELP,
+    help=_SHORT_HELP + '''
+
+Examine the cloud provider images matching the specified criteria to
+see if the most recently created - or user-specified - image has been
+tagged ready, indicating that it has passed acceptance testing.
+
+\b
+Examples:
+  See if the latest RHEL "base" image (A.K.A. AMI) in AWS has passed acceptance tests
+  $ oct image_ready --provider aws --operating_system rhel --stage base
+
+  Check if the latest centos "build" image is *not* ready in AWS, suppress non-error output (e.g. for use in a script):
+  $ oct image_ready --provider aws --operating_system centos --stage build --needs-validation --quiet
+''',
+)
+@operating_system_option
+@provider_option
+@stage_option
+@ami_id_option
+@pass_context
+@option('--needs-validation', '-n', is_flag=True, help='Returns true (0) if the newest matching image is not tagged ready.')
+@option('--quiet', '--silent', '-q', 'quiet', is_flag=True, help='Suppress standard output')
```

Comment:
We don't need both `quiet` and `silent` here

Reply:
Premature optimization?

Reply:
no?